configuration/configuration.go

```diff
@@ -0,0 +1,111 @@
+//    scheduler { storage_uri }
+//    controller { compute_ca, compute_cert, identity_user, identity_password }
+//    launcher { compute_net, mgmt_net }
+//    image_service { url }
+//    identity_service { url }
+//
+// so we need to have at least those values set in our config
+//
+// TODO: proper validation of values set in yaml setup
+func validMinConf(conf *payloads.Configure) bool {
+	return (conf.Configure.Scheduler.ConfigStorageURI != "" &&
+		conf.Configure.Controller.ComputeCACert != "" &&
+		conf.Configure.Controller.ComputeCert != "" &&
+		conf.Configure.Controller.IdentityUser != "" &&
+		conf.Configure.Controller.IdentityPassword != "" &&
+		conf.Configure.Launcher.ComputeNetwork != "" &&
+		conf.Configure.Launcher.ManagementNetwork != "" &&
+		conf.Configure.ImageService.URL != "" &&
+		conf.Configure.IdentityService.URL != "")
+}
+
+func getConfiguration(yamlConf []byte) (conf payloads.Configure, err error) {
+	err = yaml.Unmarshal(yamlConf, &conf)
+	if err != nil {
+		return conf, err
+	}
+	return
+}
+
+func getPayload(conf *payloads.Configure) (payload []byte, err error) {
+	if !validMinConf(conf) {
+		return nil, errors.New(
+			"minimal configuration is not met or yaml is malformed")
+	}
+	payload, err = yaml.Marshal(&conf)
+	if err != nil {
+		return nil, err
+	}
+	return payload, nil
+}
+
+func fillDefaults(conf *payloads.Configure) {
+	if conf.Configure.Scheduler.ConfigStorageType == "" {
+		conf.Configure.Scheduler.ConfigStorageType = payloads.Filesystem
+	}
+	if conf.Configure.Controller.ComputePort == 0 {
+		conf.Configure.Controller.ComputePort = 8774
+	}
+	if conf.Configure.ImageService.Type == "" {
+		conf.Configure.ImageService.Type = payloads.Glance
+	}
+	if conf.Configure.IdentityService.Type == "" {
+		conf.Configure.IdentityService.Type = payloads.Keystone
+	}
+}
+
+// TODO: add etcd support related scheme(s)
+func discoverDriver(uriStr string) (storageType payloads.StorageType, err error) {
+	uri, err := url.Parse(uriStr)
+	if err != nil {
+		return storageType, err
+	}
+	if uri.Scheme != "file" {
+		return storageType, fmt.Errorf(
+			"Configuration URI Scheme '%s' not supported", uri.Scheme)
+	}
+	storageType = payloads.Filesystem
+	return storageType, nil
+}
+
+// GetConfiguration returns the payload to be set
+// in the SSNTP server. It needs an URI string
+// where it can read the configuration in yaml format.
+// If the URI is not valid or the yaml is either incomplete or
+// malformed, the payload recieved will be nil and err should
+// specify what the failure is
+func GetConfiguration(uri string) (payload []byte, err error) {
+	var d driver
+	driverType, err := discoverDriver(uri)
+	//conf_yaml, err := loadFile(uri)
+	if err != nil {
+		return nil, err
+	}
+	if driverType == payloads.Filesystem {
+		d = &file{}
+	} else if driverType == payloads.Etcd {
+		d = &etcd{}
+	} else {
+		return nil, fmt.Errorf("Unable to determine configuration method for %s", uri)
+	}
```

Comment:
@sameo  What about refactoring this (L95-101) to something like:

	if driverType == payloads.Etcd {
		d = &etcd{}
	} else {
		d = &file{}
	}


That's because current 'else' implementation (L99-101) is unreachable due to return values of `discoverDriver` which is used to setup driverType variable. Don't know if also a switch structure would work for future drivers integration.